### ISSUE TITLE
Build failure OSX "cannot tap homebrew/science"

### ISSUE THREAD

**Issue by mopg** [JuliaOpt/NLopt.jl#87]
When I try to install NLopt with Julia v0.6 with `Pkg.add("NLopt")` on my Mac with version 10.12.6, I keep getting the following errors:

> julia> Pkg.build("NLopt")
> INFO: Building Homebrew
> INFO: Building NLopt
> WARNING: Compat.KERNEL is deprecated.
>   likely near /Users/Max/.julia/v0.6/NLopt/deps/build.jl:56
> WARNING: Compat.KERNEL is deprecated.
>   likely near /Users/Max/.julia/v0.6/NLopt/deps/build.jl:56
> in can_use at /Users/Max/.julia/v0.6/Homebrew/src/bindeps_integration.jl
> WARNING: Compat.KERNEL is deprecated.
>   likely near /Users/Max/.julia/v0.6/NLopt/deps/build.jl:56
> in package_available at /Users/Max/.julia/v0.6/Homebrew/src/bindeps_integration.jl
> Cloning into '/Users/Max/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science'...
> remote: Counting objects: 31882, done.
> remote: Compressing objects: 100% (13/13), done.
> remote: Total 31882 (delta 6), reused 15 (delta 5), pack-reused 31864
> Receiving objects: 100% (31882/31882), 9.25 MiB | 14.83 MiB/s, done.
> Resolving deltas: 100% (19516/19516), done.
> Error: Invalid formula: /Users/Max/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science/acado.rb
> Error: Invalid formula: /Users/Max/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science/nextflow.rb
> Error: Cannot tap homebrew/science: invalid syntax in tap!
> Cloning into '/Users/Max/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science'...
> remote: Counting objects: 31882, done.
> remote: Compressing objects: 100% (13/13), done.
> remote: Total 31882 (delta 6), reused 15 (delta 5), pack-reused 31864
> Receiving objects: 100% (31882/31882), 9.25 MiB | 9.21 MiB/s, done.
> Resolving deltas: 100% (19516/19516), done.
> Error: Invalid formula: /Users/Max/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science/acado.rb
> Error: Invalid formula: /Users/Max/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science/nextflow.rb
> Error: Cannot tap homebrew/science: invalid syntax in tap!

Any thoughts on what I can do to fix this? I feel like it is similar to #57, but those fixes did not help here. My Homebrew package is at the latest version.

**Comment by EthanAnderes** [comment]
I'm not sure this is the same problem, but I just ran into this today:

```
INFO: Building NLopt
WARNING: Compat.KERNEL is deprecated.
  likely near /Users/ethananderes/.julia/v0.6/NLopt/deps/build.jl:56
WARNING: Compat.KERNEL is deprecated.
  likely near /Users/ethananderes/.julia/v0.6/NLopt/deps/build.jl:56
in can_use at /Users/ethananderes/.julia/v0.6/Homebrew/src/bindeps_integration.jl
WARNING: Compat.KERNEL is deprecated.
  likely near /Users/ethananderes/.julia/v0.6/NLopt/deps/build.jl:56
in package_available at /Users/ethananderes/.julia/v0.6/Homebrew/src/bindeps_integration.jl
Cloning into '/Users/ethananderes/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science'...
remote: Counting objects: 32203, done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 32203 (delta 11), reused 20 (delta 7), pack-reused 32176
Receiving objects: 100% (32203/32203), 9.59 MiB | 7.81 MiB/s, done.
Resolving deltas: 100% (19713/19713), done.
Warning: Use nlopt instead of deprecated homebrew/science/nlopt
============================================[ ERROR: NLopt ]============================================

LoadError: SystemError: opening file /Users/ethananderes/.julia/v0.6/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science/nlopt.rb: No such file or directory
while loading /Users/ethananderes/.julia/v0.6/NLopt/deps/build.jl, in expression starting on line 56

========================================================================================================

============================================[ BUILD ERRORS ]============================================

WARNING: NLopt had build errors.

 - packages with build errors remain installed in /Users/ethananderes/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("NLopt")`
 - build a single package by running its `deps/build.jl` script

========================================================================================================
```

**Comment by pearlzli** [comment]
I'm having the same problem as @EthanAnderes while running a Travis build (for a repo which has NLopt as a dependency):

```
INFO: Building NLopt
Warning: Use nlopt instead of deprecated homebrew/science/nlopt
================================[ ERROR: NLopt ]================================
LoadError: SystemError: opening file /Users/travis/.julia/v0.5/Homebrew/deps/usr/Library/Taps/homebrew/homebrew-science/nlopt.rb: No such file or directory
while loading /Users/travis/.julia/v0.5/NLopt/deps/build.jl, in expression starting on line 56
```

**Comment by stevengj** [comment]
Maybe change [this line](https://github.com/JuliaOpt/NLopt.jl/blob/78a635d334b51dd6ab696bb7d2670973e794846e/deps/build.jl#L19) to a different tap name?   Just `nlopt`?

**Comment by EthanAnderes** [comment]
Yeah, the following workaround did the trick for me:

```
julia> using Homebrew
julia> Homebrew.add("nlopt")
julia> Pkg.build("NLopt")
```

**Comment by stevengj** [comment]
A patch for the NLopt build script would be welcome.

**Comment by stevengj** [comment]
Oh, nevermind, was already fixed in #85.
